packages/math/is_prime.pony

```diff
@@ -0,0 +1,80 @@
+primitive IsPrime[A: (Integer[A] val & Unsigned)]
+  """Primality test using 6k+-1 optimization."""
```

Comment:
Is there anything that can be linked to that explains this optimization?

Reply:
I am trying to find a good source. My search landed me on Wikipedia, but I would prefer something more definitive.

Reply:
The relevant passage is this, but has no source to it.

> We can improve this method further. Observe that all primes greater than 3 are of the form 6k ± 1, where k is any integer greater than 0. This is because all integers can be expressed as (6k + i), where i = −1, 0, 1, 2, 3, or 4. Note that 2 divides (6k + 0), (6k + 2), and (6k + 4) and 3 divides (6k + 3). So, a more efficient method is to test whether n is divisible by 2 or 3, then to check through all numbers of the form 6 k   ±   1 ≤ n {\displaystyle \scriptstyle 6k\ \pm \ 1\leq {\sqrt {n}}} {\displaystyle \scriptstyle 6k\ \pm \ 1\leq {\sqrt {n}}}. This is 3 times faster than testing all numbers up to √n.

Reply:
Anything is better than nothing! (Maybe) 😉